apps/docs/src/guide/predicates/methods.md

```diff
@@ -0,0 +1,55 @@
+# Interacting With Predicates
+
+The `Predicate` class extends the [`Account`](https://docs.fuel.network/docs/fuels-ts/account/) class, inheriting all its methods. Therefore, there are multiple ways to interact with predicates, but broadly speaking, we can think about three:
+ - `Checking Balances`
+ - `Transactions`
+ - `Transfers`
+
+
+## Checking Balances
+
+### `getBalances`
+
+This will return the balances of all assets owned by the predicate.
+
+<<< @/../../docs-snippets/src/guide/wallets/checking-balances.test.ts#checking-balances-2{ts:line-numbers}
+
+### `getResourcesToSpend`
+
+This will return the resources owned by a predicate so that they can be added to a transaction request.
+
+This method is called under the hood when using [`transfer`](./methods.md#transfer) or [`createTransfer`](./methods.md#createtransfer).
+
+You may want to use this method when using a predicate in an existing transaction request.
+
+<<< @/../../docs-snippets/src/guide/predicates/interacting-with-predicates.test.ts#interacting-with-predicates-1{ts:line-numbers}
+
+## Transactions
+
+### `sendTransaction`
+
+This is used to send a transaction to the node.
+
+<<< @/../../docs-snippets/src/guide/predicates/interacting-with-predicates.test.ts#interacting-with-predicates-2{ts:line-numbers}
+
+### `simulateTransaction`
+
+The `simulateTransaction` method should be used to dry-run a predicate call, ensuring that there are sufficient funds to cover the transaction fees, without consuming any resources.
+
+<<< @/../../docs-snippets/src/guide/predicates/interacting-with-predicates.test.ts#interacting-with-predicates-3{ts:line-numbers}
+
+## Transfers
+
+### `createTransfer`
+
+The `createTransfer` method creates a transfer request and add the transfer details to that request. This method estimates the transaction cost via a dry-run, and then funds it with predicate resources, which is useful since the returned transaction request can now be submitted with greater certainty that it will succeed.
+
+ This can be useful should you need to modify the transfer request or use its properties before sending it to the node.
```

Comment:
```suggestion
The `createTransfer` method creates a transaction request with all the necessary transfer details. It automatically estimates the transaction costs via a dry-run call and funds the request with the required predicate resources. After this, one can submit the returned transaction request with greater certainty that it will succeed.

However, please remember that you can still modify the transfer request details and use its properties before submitting it to the node.
```

I'd also say these two lines seem irrelevant to the purpose of the code snippet and could be deleted:

```ts
const chainId = provider.getChainId();

const txId = transactionRequest.getTransactionId(chainId);
```

Reply:
My apologies, when I applied the suggestion you made [here](https://github.com/FuelLabs/fuels-ts/pull/2345/commits/e4e78bb58cd57c741b85d1273d7960aa0f2d3a40) Github automatically resolved the issue, and so that consideration eluded me. 

This code snippet is re-used from another [guide](https://github.com/FuelLabs/fuels-ts/blob/mc/docs/improve-predicate-docs/apps/docs/src/guide/predicates/send-and-spend-funds-from-predicates.md#L45), but I agree with your point, I'll implement a new snippet for this.

Reply:
Perhaps you can just move the two lines below where the snippet ends.

Reply:
See [`2a8de44` (#2345)](https://github.com/FuelLabs/fuels-ts/pull/2345/commits/2a8de44dc130234cfff5cc6b455c9ad6f1be188f)

Reply:
TBH, the text preceding this snippet could be clearer as well. 🙂 

We could move the last line of text inside the snippet as a comment.

Both cases would benefit from it, and we could share the same snippet.

See what you think.

![suggestion](https://github.com/FuelLabs/fuels-ts/assets/26660/688a8681-2870-41db-a457-d3f725a0cf73)

Reply:
Ooof, I didn't see your message before sending mine.

The UI may not refresh while one is writing a reply. 😓

Reply:
> Ooof, I didn't see your message before sending mine.
> 
> The UI may not refresh while one is writing a reply. 😓

Happens to me quite frequently as well, I end up missing comments at times 😢 




> TBH, the text preceding this snippet could be clearer as well. 🙂
> 
> We could move the last line of text inside the snippet as a comment.
> 
> Both cases would benefit from it, and we could share the same snippet.
> 
> See what you think.
> 

I agree, and it promotes code reuse so I will make that change.

Reply:
See [`80af938` (#2345)](https://github.com/FuelLabs/fuels-ts/pull/2345/commits/80af938e61d654730c281f02f729725b79f97b10)

Reply:
Nice! Not sure, but then I guess you'll need to undo [this](https://github.com/FuelLabs/fuels-ts/pull/2345/commits/2a8de44dc130234cfff5cc6b455c9ad6f1be188f#diff-bf9cefcebbc4d0127bda5f3a965bcb75c3d6e4130467325e4701ed38d0b91dd1R122-R156) as well I?

Anyway, I am leaving this to you and resolving it for now.

Reply:
Correct 😅